File: src/main/java/mods/eln/ghost/GhostManagerNbt.java
package mods.eln.ghost;

import mods.eln.Eln;
import mods.eln.Vars;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.world.WorldSavedData;

public class GhostManagerNbt extends WorldSavedData {
    public GhostManagerNbt(String par1Str) {
        super(par1Str);
    }

    @Override
    public boolean isDirty() {
        return true;
    }

    @Override
    public void readFromNBT(NBTTagCompound nbt) {
        Vars.ghostManager.loadFromNBT(nbt);
    }

    @Override
    public void writeToNBT(NBTTagCompound nbt) {
        //Eln.ghostManager.saveToNbt(nbt, Integer.MIN_VALUE);
    }
}

File: src/main/kotlin/mods/eln/ghost/GhostManagerNbt.kt
package mods.eln.ghost

import mods.eln.Eln
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.world.WorldSavedData

class GhostManagerNbt(par1Str: String?) : WorldSavedData(par1Str) {
    override fun isDirty(): Boolean {
        return true
    }

    override fun readFromNBT(nbt: NBTTagCompound) {
        Eln.ghostManager.loadFromNBT(nbt)
    }

    override fun writeToNBT(nbt: NBTTagCompound) {}
}
